openedx/core/djangoapps/discussions/api/data.py

```diff
@@ -0,0 +1,25 @@
+import attr
+from opaque_keys.edx.keys import CourseKey
+
+
+class ObjectDoesNotExist(Exception):
+    """
+    Imitating Django model conventions, we put a subclass of this in some of our
+    data classes to indicate when something is not found.
+    """
+
+
+@attr.s(frozen=True)
+class CourseDiscussionConfigData:
+    """
+    Course Discussion Configuration Data Object
+    """
+
+    course_key = attr.ib(type=CourseKey)
+    enabled = attr.ib(type=bool)
+    lti_configuration = attr.ib(type=dict)
+    plugin_configuration = attr.ib(type=dict)
+    provider = attr.ib(type=str)
```

Comment:
Random thought - in the future, if we get our hard-coded list out of code and into the database, this will likely become a foreign key.

Reply:
Perhaps `provider_type` would be clearer?

Reply:
Done.